OSDOCS#13454: Updates to the web console tutorial

bergerhoffer opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
4.19

Issue:
https://issues.redhat.com/browse/OSDOCS-13254

Link to docs preview:
https://91212--ocpdocs-pr.netlify.app/openshift-enterprise/latest/tutorials/dev-app-web-console

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

~~Will need #90215 to merge first before can make changes to the web console assembly.~~ Done

**Note to reviewer:** Some of the wording updates match what was already merged for the CLI version of the tutorial in #90215.

bergerhoffer committed:
OSDOCS#13454: Updates to the web console tutorial

bergerhoffer commented on modules/getting-started-web-console-creating-new-project.adoc:
```diff
@@ -1,33 +1,25 @@
-* You are logged in to the {product-title} web console.
-* You are in the *Developer* perspective.
 * You have the appropriate roles and permissions in a project to create applications and other workloads in {product-title}.
+// TODO: This requirement isn't in the CLI version. Is it necessary?
```
@AlexonOliveiraRH I think we can remove the above line ("You have the appropriate roles..."). It's not in the CLI version. wdyt?

AlexonOliveiraRH replied:
> @AlexonOliveiraRH I think we can remove the above line ("You have the appropriate roles..."). It's not in the CLI version. wdyt?

@bergerhoffer although this is not CLI, permissions are bound across all clusters resources, independently of which platform you're using, i.e. CLI or web console, the user still needs to guarantee the proper permissions.

bergerhoffer replied:
@AlexonOliveiraRH What permissions should we say that they need to have then? (And if we add it here, we should also add that to the CLI version)

bergerhoffer replied:
Hm, this still doesn't seem to check out to me, because if they are the ones that created the project, then shouldn't they automatically get whatever permissions needed to work in that project?

The only thing I could see saying here is about whether someone has access to the cluster, but doesn't have permission to create a project (if there is such a role like that). wdyt?

AlexonOliveiraRH replied:
> Hm, this still doesn't seem to check out to me, because if they are the ones that created the project, then shouldn't they automatically get whatever permissions needed to work in that project?
> 
> The only thing I could see saying here is about whether someone has access to the cluster, but doesn't have permission to create a project (if there is such a role like that). wdyt?

@bergerhoffer for example, the same user could get only a `view` permission, so this user would be able to navigate into the project, see things, but wouldn't be able to create nothing, neither change anything. That's why the cluster admin should grant this user at least `edit` permission (or `admin` in the namespace scope only) to create the project and do the rest of the tutorial

bergerhoffer commented on modules/getting-started-web-console-deploying-python-app.adoc:
```diff
@@ -1,48 +1,66 @@
-====
-If the detected builder image is Dockerfile, select *Edit Import Strategy*. Select *Builder Image* and then click *Python*.
-====
+// TODO: If it never detects it properly, update these steps to just always go through them (Remove the "If...")
```
@AlexonOliveiraRH When I tested it, it didn't detect Builder Image / Python, it detected Dockerfile. When you test, can you see if yours also detects Dockerfile?

If that's what it's always going to detect, and they are always going to need to change it, I'll probably update this text some (line 20 and the "If..." on line 24).

AlexonOliveiraRH replied:
> @AlexonOliveiraRH When I tested it, it didn't detect Builder Image / Python, it detected Dockerfile. When you test, can you see if yours also detects Dockerfile?
> 
> If that's what it's always going to detect, and they are always going to need to change it, I'll probably update this text some (line 20 and the "If..." on line 24).

@bergerhoffer yep, mine detected `Dockerfile` as well. We need to change it everytime. Go ahead with the suggested edit, please.

bergerhoffer commented on modules/getting-started-web-console-load-data-output.adoc:
```diff
@@ -32,8 +32,7 @@ Before loading the data, add the proper labels to the `mongodb-nationalparks` an
-image::getting-started-map-national-parks.png[]
+// TODO: Give an example of the URL.
+
+// The CLI version just uses `http://localhost:8080/ws/data/load`. Could we use that instead? Test it out
```
@AlexonOliveiraRH This web console version has us getting the actual URL and then adding `/ws/data/load` to it. But the CLI version just had them go to `http://localhost:8080/ws/data/load`.

I didn't get a chance to test it today, but do you think the web console users could also just use this URL instead of their cluster-specific one?

AlexonOliveiraRH replied:
> @AlexonOliveiraRH This web console version has us getting the actual URL and then adding `/ws/data/load` to it. But the CLI version just had them go to `http://localhost:8080/ws/data/load`.
> 
> I didn't get a chance to test it today, but do you think the web console users could also just use this URL instead of their cluster-specific one?

@bergerhoffer it won't work the same with the web console version because when it run it through the CLI, we run this load command directly into the application pod, which doesn't need to go outside, like through a Service or a Route, to access the application URL, since the application is running locally (inside the pod itself). But to be able to do it through the web console, we need to access the application through an external URL exposed through a Route, that's why wee need to append this `/ws/data/load` to the URL provided in the web console. The alternative to use the `http://localhost:8080/ws/data/load` command as we do in the CLI version would be accessing the pod's terminal through the web console and then run the command inside the pod, which is basically what we do in the CLI version, but IMHO that dismiss a little bit the use of the web console features.

AlexonOliveiraRH commented on modules/getting-started-web-console-granting-permissions.adoc:
```diff
@@ -1,30 +1,30 @@
-. Navigate to *User Management* and then click *RoleBindings*.
+. Navigate to *User Management* -> *RoleBindings*.
 . Click *Create binding*.
 . Select `Namespace role binding (RoleBinding)`.
```
@bergerhoffer I couldn't find this specific option. Where did you find it?

bergerhoffer replied:
Please double check, I saw this option in the UI, but let me know if you don't see it when you try with 4.19

AlexonOliveiraRH commented on modules/getting-started-web-console-deploying-first-image.adoc:
```diff
@@ -1,32 +1,45 @@
 .Prerequisites
 
-* You are logged in to the {product-title} web console.
-* You are in the *Developer* perspective.
```
@bergerhoffer We should keep both these pre-reqs, otherwise the user won't understand in which perspective should they be logged to execute the following steps.

bergerhoffer replied:
Ignoring per our slack convo, since 4.19 got rid of these perspectives

AlexonOliveiraRH commented on modules/getting-started-web-console-creating-new-project.adoc:
```diff
@@ -1,33 +1,25 @@
 .Prerequisites
 
-* You are logged in to the {product-title} web console.
-* You are in the *Developer* perspective.
```
@bergerhoffer this specific step requires the _Administrator_ perspective, while the next steps would require the _Developer_ perspective, so I'd get these pre-reqs back only changing the `Developer` perspective to the `Administrator` perspective.

bergerhoffer replied:
Ignoring per our slack convo, since 4.19 got rid of these perspectives

AlexonOliveiraRH commented:
@bergerhoffer well done! Everything worked like a charm. With the exception of some minor tweaking you may do, LGTM to move forward.

bergerhoffer commented:
/label peer-review-needed

kcarmichael08 commented on tutorials/dev-app-web-console.adoc:
```diff
@@ -6,41 +6,67 @@ include::_attributes/common-attributes.adoc[]
 [role="_additional-resources"]
 .Additional resources
-* xref:../authentication/using-rbac.adoc#default-roles_using-rbac[Default cluster roles]
 * xref:../applications/projects/working-with-projects.adoc#viewing-a-project-using-the-web-console_projects[Viewing a project using the web console]
```
```suggestion
* xref:../applications/projects/working-with-projects.adoc#viewing-a-project-using-the-web-console_projects[Viewing a project by using the web console]
```

kcarmichael08 commented on modules/getting-started-web-console-connecting-a-database.adoc:
```diff
@@ -1,32 +1,30 @@
-. Select *Deployment* as the *Resource*.
-. Unselect the checkbox next to *Create route to the application*.
-. In the *Advanced Options* section, click *Deployment* to add environment variables to add the following environment variables:
+. From the *Quick create* (image:fa-plus-circle.png[title="Quick create menu"]) drop-down menu in the upper right corner, click *Container images*.
```
I think "drop-down" can be omitted per IBM Style for "menu" and "list" guidance where it seems to omit "drop-down" for both:

Menu: "Label: Refer to a menu or menu item by its label only, unless menu or menu item is required for clarity."
Drop-down list: "Refer to a drop-down list by its label, followed by list."

kcarmichael08 commented on modules/getting-started-web-console-deploying-first-image.adoc:
```diff
@@ -1,32 +1,38 @@
+. Select *Image name from external registry* and enter `quay.io/openshiftroadshow/parksmap:latest`.
+. Scroll to the *General* section.
+. In the *Application name* field, enter `national-parks-app`.
+. In the *Name* field, ensure the value is `parksmap`.
```
```suggestion
. In the *Name* field, ensure that the value is `parksmap`.
```

kcarmichael08 commented on modules/getting-started-web-console-deploying-first-image.adoc:
```diff
@@ -1,32 +1,38 @@
-. Select *Deployment* as the *Resource*.
-. Select *Create route to the application*.
-. In the *Advanced Options* section, click *Labels* and add labels to better identify this deployment later. Labels help identify and filter components in the web console and in the command line. Add the following labels:
+. From the *Quick create* (image:fa-plus-circle.png[title="Quick create menu"]) drop-down menu in the upper right corner, click *Container images*.
```
see comment about "drop-down"

kcarmichael08 commented on modules/getting-started-web-console-examining-pod.adoc:
```diff
@@ -1,41 +1,34 @@
+{product-title} uses the Kubernetes concept of a _pod_, which is one or more containers deployed together on one host, and the smallest compute unit that can be defined, deployed, and managed.
 Pods are the rough equivalent of a machine instance, physical or virtual, to a container.
 
 The *Overview* panel enables you to access many features of the `parksmap` deployment. The *Details* and *Resources* tabs enable you to scale application pods, check build status, services, and routes.
```
```suggestion
The *Overview* panel enables you to access many features of the `parksmap` deployment. The *Details* and *Resources* tabs enable you to scale application pods and check build status, services, and routes.
```

bergerhoffer replied:
Tweaked

kcarmichael08 commented on modules/getting-started-web-console-deploying-python-app.adoc:
```diff
@@ -1,48 +1,58 @@
-.. `component=nationalparks`
-.. `role=backend`
-.. `type=parksmap-backend`
+. In the *Application* field, ensure the value is `national-parks-app`.
```
```suggestion
. In the *Application* field, ensure that the value is `national-parks-app`.
```

kcarmichael08 commented on modules/getting-started-web-console-deploying-python-app.adoc:
```diff
@@ -1,48 +1,58 @@
-[x-]`https://github.com/openshift-roadshow/nationalparks-py.git`
-+
-A builder image is automatically detected.
+. From the *Quick create* (image:fa-plus-circle.png[title="Quick create menu"]) drop-down menu in the upper right corner, click *Import from Git*.
```
see other comment re: "drop-down"

kcarmichael08 commented on modules/getting-started-web-console-connecting-a-database.adoc:
```diff
@@ -38,5 +36,16 @@ Once you mark the `national-parks-app` application as a backend for the map visu
+Click *Add value* to add each additional environment variable.
+====
+
+. In the *Advanced options* section, deselect *Create a route*.
```
```suggestion
. In the *Advanced options* section, clear *Create a route*.
```
(if appropriate) - If this is a checkbox, per IBM, "Use clear to refer to deselecting the choice."

kcarmichael08 commented on modules/getting-started-web-console-creating-secret.adoc:
```diff
@@ -1,43 +1,52 @@
-The `Secret` object provides a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, private source repository credentials, and so on.
-Secrets decouple sensitive content from the pods. You can mount secrets into containers using a volume plugin or the system can use secrets to perform actions on behalf of a pod.
-The following procedure adds the secret `nationalparks-mongodb-parameters` and mounts it to the `nationalparks` workload.
+The `nationalparks` application needs information, such as the database name, username, and passwords, on how to access the MongoDB database. However, because this information is sensitive, you should not store it directly in the pod.
```
```suggestion
The `nationalparks` application needs information, such as the database name, username, and passwords, to access the MongoDB database. However, because this information is sensitive, you should not store it directly in the pod.
```

kcarmichael08 commented on modules/getting-started-web-console-creating-secret.adoc:
```diff
@@ -1,43 +1,52 @@
+
+You can use a _secret_ to store sensitive information, and share that secret with workloads.
+
+`Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, private source repository credentials, and so on. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin, or the system can use secrets to perform actions on behalf of a pod.
```
```suggestion
`Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, and private source repository credentials. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin. The system can then use secrets to perform actions on behalf of a pod.
```
Suggested this edit to the last sentence because they seem like two separate things that are sequential...? But I'm not very knowledgeable about this, so ignore if n/a.

bergerhoffer replied:
Good question. This was already here, but I agree that this doesn't seem to make total sense. It looks like it was taken directly from the "Understanding secrets" section here: https://91212--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets#nodes-pods-secrets-about_nodes-pods-secrets

@AlexonOliveiraRH Any thoughts on how we could improve this wording? Should it be saying that you can either mount the secret by using the volume plugin or pass it as env vars (like we're doing here)? And then I'm not sure whether that last sentence ("The system can then use secrets to perform actions on behalf of a pod.") makes sense afterward? wdyt?

```suggestion
`Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, and private source repository credentials. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin or by passing the secret in as an environment variable. The system can then use secrets to perform actions on behalf of a pod.
```

bergerhoffer replied:
@AlexonOliveiraRH do you mean the way it originally was was correct?

> You can mount secrets into containers by using a volume plugin, or the system can use secrets to perform actions on behalf of a pod.

Or the way I've updated it to be now is correct?

> You can mount secrets into containers by using a volume plugin or by passing the secret in as an environment variable. The system can then use secrets to perform actions on behalf of a pod.

kcarmichael08 commented on modules/getting-started-web-console-creating-secret.adoc:
```diff
@@ -1,43 +1,52 @@
 . Click *Add Secret to workload*.
-.. From the drop down menu, select `nationalparks` as the workload to add.
-.. Click *Save*.
+. From the *Add this secret to workload* drop-down list, select `nationalparks`.
```
```suggestion
. From the *Add this secret to workload* list, select `nationalparks`.
```

kcarmichael08 reviewed (commented):
This looks great! Had a few suggestions/questions but nothing big.

AlexonOliveiraRH commented:
> @AlexonOliveiraRH This is ready for another review.
> 
> If it helps, here's the diff of what I changed since the last time you reviewed: https://github.com/openshift/openshift-docs/compare/71e0935bd6792e1fca1d06db0995474cebe4d99e..edd229dc788dea603795722c4fc2b4a6f7b3c7ed
> 
> The main updates I'd want you to check are
> 
> 1. The rewording of the secrets paragraph (the only open comment from above)
> 2. I ended up adding something to the tutorial prereqs about the necessary permissions needed to create projects and add to them. Please check that wording and see if that makes sense. I mentioned cluster-admin and edit roles as examples of what they'd need.

@bergerhoffer these changes LGTM, thank you. The only it needs to be fixed is exactly the permission part. Please, if you could change from `cluster-admin` to only `admin` and explain that this only into the namespace scope, because `cluster-admin` is a wide permission, cluster scope.

bergerhoffer commented:
@AlexonOliveiraRH Okay here are my latest updates after our last discussions in Slack. I also went through and tested using Developer Sandbox, and found a few more little tweaks to make.

* Added to skip creating a project if using Developer Sandbox
* Noted places where the project name might be different if they're using Developer Sandbox or another project name
* Mentioned the roles required. I removed 'edit' as an option for the project-level perms, because they need at least 'admin' to grant the view permissions
* The secrets wording we talked about
* Found a few typos where `parksmap` should have been `nationalparks`,  and made a few other small tweaks
* Mentioned that the browser may ask for their location, just so they're not surprised, so they know what it does.

Here's the diff if it helps: https://github.com/openshift/openshift-docs/compare/edd229dc788dea603795722c4fc2b4a6f7b3c7ed..1d43fbcf05161eeb56fbbf0c0c2377e6a9f89b6d

Let me know how these updates look, thanks!

AlexonOliveiraRH commented:
> @AlexonOliveiraRH Okay here are my latest updates after our last discussions in Slack. I also went through and tested using Developer Sandbox, and found a few more little tweaks to make.
> 

Nice.

> * Added to skip creating a project if using Developer Sandbox

Good catch.

> * Noted places where the project name might be different if they're using Developer Sandbox or another project name

Good catch too.

> * Mentioned the roles required. I removed 'edit' as an option for the project-level perms, because they need at least 'admin' to grant the view permissions

Sounds good to me.

> * The secrets wording we talked about

Perfect.

> * Found a few typos where `parksmap` should have been `nationalparks`,  and made a few other small tweaks

Also a good catch.

> * Mentioned that the browser may ask for their location, just so they're not surprised, so they know what it does.
> 

Nice touch.

> Here's the diff if it helps: https://github.com/openshift/openshift-docs/compare/edd229dc788dea603795722c4fc2b4a6f7b3c7ed..1d43fbcf05161eeb56fbbf0c0c2377e6a9f89b6d
> 
> Let me know how these updates look, thanks!

I just finished my review and kudos to you, ma'am. It looks very good to me. :) I think you addressed all the main points we discussed, specially the recent ones.

bergerhoffer commented:
@yapei Here is the PR to update the web console tutorial. Can you PTAL when you get a chance?

Preview: https://91212--ocpdocs-pr.netlify.app/openshift-enterprise/latest/tutorials/dev-app-web-console

Similar changes were made to the CLI version of the tutorial in #90215, which has been approved/merged. This PR also makes a few further tweaks to the CLI version that we made after working on this web console version. Let me know if you have any questions, thanks!

yapei commented on modules/getting-started-web-console-granting-permissions.adoc:
```diff
@@ -1,30 +1,40 @@
-. Navigate to *User Management* and then click *RoleBindings*.
+. Navigate to *User Management* -> *RoleBindings*.
 . Click *Create binding*.
 . Select `Namespace role binding (RoleBinding)`.
```
`Namespace role binding (RoleBinding)` option is only visible/accessible for cluster admin users 
this is the page project-level admin user will see after click `Create binding`
![Screenshot 2025-04-10 at 1 54 14 PM](https://github.com/user-attachments/assets/5dd52ca6-a6db-4c3f-be3b-3eeb07dabd40)

This is the page cluster admin user sees
![Screenshot 2025-04-10 at 1 52 17 PM](https://github.com/user-attachments/assets/ffa06016-5e0b-45b4-9109-779a022cad32)

bergerhoffer replied:
Thanks! I think I'll just remove that step though because it is already selected by default for cluster admins, so there really isn't any action that needs to be taken here.

yapei commented on modules/getting-started-web-console-load-data-output.adoc:
```diff
@@ -1,39 +1,41 @@
-* You are logged in to the {product-title} web console.
-* You are in the *Developer* perspective.
-* You have a deployed image.
+* You are in the *Developer* perspective of the web console.
```
we disabled `Developer` perspective by default in OCP 4.19, `Workloads -> Topology` is moved to `Administrator` perspective though

bergerhoffer replied:
Good catch, I forgot to remove it from this one.

yapei commented:
@bergerhoffer  Changes look perfect from QE's point of view, some tiny update requests above, PTAL

bergerhoffer commented:
@yapei I've updated the two items you pointed out, if you can take another look please. Thanks!

yapei commented:
Thanks @bergerhoffer LGTM
